pages/glossary/label-in-name-algorithm.md

```diff
@@ -0,0 +1,39 @@
+---
+title: Label in Name Algorithm
+key: label-in-name-algorithm
+unambiguous: true
+objective: false
+input_aspects:
+  - CSS styling
+  - DOM tree
+---
+
+To check whether an [element][] has <dfn>its label contained in its name</dfn>, follow this algorithm:
+
+Let 'label' be the [visible inner text][] of the target element.  Let 'name' be the [accessible name][] of the target element.  Both 'label' and 'name' are strings.
+
+Sub-algorithm to tokenize a string:
+
+- Convert the string to lower case.
+- For each character that either a) represents non-text content, or b) isn't a letter or a digit: replace that character with a space character.
```

Comment:
I think this causes false positives with words that are not consistently hyphenated. For example:

'antimatter' vs 'anti-matter' (the [Wikipedia antimatter](https://en.wikipedia.org/wiki/Antimatter) article uses both)
'derisk' vs 're-risk' (Cambridge dictionary uses first spelling, Collins dictionary uses the second)
'nonnegative' vs 'non-negative' https://math.stackexchange.com/a/3344027

Reply:
You're right, and I assume the voice control software would handle it fine, but if it was a customer who was arguing this, I would say "too bad" (diplomatically, of course).   This is part of a larger issue that I call "you need to play ball with the automated check".   I run into customer arguments like this often enough that I have a "canned response" it.  It reads, in part: "It's not feasible for an automated checker to know any better in this situation.  When you use an automated checker instead of a manual audit, in order to gain efficiency, you lose accuracy.  This is an unavoidable example of that." 

For the page author to "play ball" for "label in name" would be easy: use the same spelling, including hyphenation, in the aria-label that they used in the visible label.  This is, I think, not a lot to ask of a page author.

I could add an assumption to the rule.  Something along the lines of "This rule assumes that for any word (including any hyphenated word) that appears in both the accessible name and the visible label, the same spelling and hyphenation is used in both places.  For example, using 'antimatter' in the accessible name and 'anti-matter' in the visible label would fail this rule, but arguably pass the Success Criterion.

Reply:
Bearing in mind this is for voice control, would a voice control user pronounce the hyphenated and non-hyphenated versions differently? Personally, I would pronounce both versions identically:

'antimatter' vs 'anti-matter' 
'derisk' vs 'de-risk' 
'nonnegative' vs 'non-negative'